tests/ManageCourses.Tests/UnitTesting/Client/ManageCoursesApiClientTest.cs

```diff
@@ -0,0 +1,190 @@
+                new HttpResponseMessage() {
+                    StatusCode = HttpStatusCode.OK,
+                    Content = payloadStringContent
+                })
+            .Verifiable();
+        }
+        private void SetupGetUrlVerification<T>(string url) where T : new()
+        {
+            var payloadJson = JsonConvert.SerializeObject(new T());
+
+            var payloadStringContent = new StringContent(payloadJson, Encoding.UTF8, "application/json" );
+
+            var setup = mockHttp.Setup(x => x.GetAsync(It.Is<Uri>(y => y.AbsoluteUri == url)));
+            setup.ReturnsAsync(
+                new HttpResponseMessage() {
+                    StatusCode = HttpStatusCode.OK,
+                    Content = payloadStringContent
+                })
+            .Verifiable();
+        }
+
+        [Test]
+        public void Courses_GetAsync()
+        {
+            var instCode ="instCode";
+            var ucasCode = "ucasCode";
+
+            var controller = "courses";
+            var leaf = $"/{instCode}/course/{ucasCode}";
+            SetupGetUrlVerification<Course>($"{baseurl}/api/{controller}{leaf}");
+
+            var result = sut.Courses_GetAsync(instCode, ucasCode).Result;
+
+            result.Should().BeOfType<Course>();
+            mockHttp.VerifyAll();
+        }
+
+        [Test]
+        public void Courses_GetAllAsync()
+        {
+            var instCode ="instCode";
+
+            var controller = "courses";
+            var leaf = $"/{instCode}";
+            SetupGetUrlVerification<InstitutionCourses>($"{baseurl}/api/{controller}{leaf}");
+
+            var result = sut.Courses_GetAllAsync(instCode).Result;
+
+            result.Should().BeOfType<InstitutionCourses>();
+            mockHttp.VerifyAll();
+        }
+
+        [Test]
+        public void Organisations_GetAsync()
+        {
+            var instCode ="instCode";
+
+            var controller = "organisations";
+            var leaf = $"/{instCode}";
+            SetupGetUrlVerification<UserOrganisation>($"{baseurl}/api/{controller}{leaf}");
+
+            var result = sut.Organisations_GetAsync(instCode).Result;
+
+            result.Should().BeOfType<UserOrganisation>();
+            mockHttp.VerifyAll();
+        }
+        [Test]
+        public void Organisations_GetAllAsync()
+        {
+            var controller = "organisations";
+            var leaf = $"";
+            SetupGetUrlVerification<List<UserOrganisation>>($"{baseurl}/api/{controller}{leaf}");
+
+            var result = sut.Organisations_GetAllAsync().Result;
+
+            result.Should().AllBeAssignableTo<IEnumerable<UserOrganisation>>();
+            mockHttp.VerifyAll();
+        }
+
+        public void Publish_PublishCourseToSearchAndCompareAsync()
+        {
+            var instCode = "instCode";
+            var courseCode = "courseCode";
+            var controller = "publish";
+            var leaf = $"/course/{instCode}/{courseCode}";
```

Comment:
I'd inline all the leaf vars in the file

Reply:
superfluous

try building the same url thats is either built by convention or via route or via httpget or via httppost, twice over

Then you recognise the need to use a template like `$"{baseurl}/api/{controller}{leaf}"`